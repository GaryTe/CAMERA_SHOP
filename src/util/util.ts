import {MutableRefObject, KeyboardEvent} from 'react';
import { Reviews, listCamera } from '../types/types-response/types-response';
import { Category, TypeCamera, LevelProduct } from '../const/const';

import dayjs from 'dayjs';


export const gettingTranslatedDate = (date: string) => dayjs (date).format ('D MMMM');


export const sortReviews = (reviews: Reviews) => {
  let resultSort: Reviews = [];

  if(reviews.length === 1) {
    resultSort = reviews;
  } else {
    resultSort = reviews.slice ().sort ((a,b) => {
      if (a.createAt < b.createAt) {
        return -1;
      }
      if (a.createAt > b.createAt) {
        return 1;
      }
      return 0;
    }).reverse();
  }

  return resultSort;
};


export const down = (products: listCamera, payload: string) => {
  let resultSort: listCamera = [];

  resultSort = products.slice ().sort ((a,b) => {
    if (a[payload] < b[payload]) {
      return -1;
    }
    if (a[payload] > b[payload]) {
      return 1;
    }
    return 0;
  }).reverse();

  return resultSort;
};

export const up = (products: listCamera, payload: string) => {
  let resultSort: listCamera = [];

  resultSort = products.slice ().sort ((a,b) => {
    if (a[payload] < b[payload]) {
      return -1;
    }
    if (a[payload] > b[payload]) {
      return 1;
    }
    return 0;
  });

  return resultSort;
};


export const filteringByCategory = (products: listCamera, payload: string): listCamera => {
  let filteredListProduct: listCamera = [];

  switch (payload) {
    case Category.Camera:
      filteredListProduct = products.slice().filter((camera) => camera.category === Category.Camera);
      break;
    case Category.Videocamera:
      filteredListProduct = products.slice().filter((Videocamera) => Videocamera.category === Category.Videocamera);
      break;
    case TypeCamera.Digital:
      filteredListProduct = products.slice().filter((camera) => camera.type === TypeCamera.Digital);
      break;
    case TypeCamera.Instant:
      filteredListProduct = products.slice().filter((camera) => camera.type === TypeCamera.Instant);
      break;
    case TypeCamera.Membranous:
      filteredListProduct = products.slice().filter((camera) => camera.type === TypeCamera.Membranous);
      break;
    case TypeCamera.Collection:
      filteredListProduct = products.slice().filter((camera) => camera.type === TypeCamera.Collection);
      break;
    case LevelProduct.Zero:
      filteredListProduct = products.slice().filter((camera) => camera.level === LevelProduct.Zero);
      break;
    case LevelProduct.Professional:
      filteredListProduct = products.slice().filter((camera) => camera.level === LevelProduct.Professional);
      break;
    case LevelProduct.Amateur:
      filteredListProduct = products.slice().filter((camera) => camera.level === LevelProduct.Amateur);
      break;
  }

  return filteredListProduct;
};

export const filteringByPrice = (
  products: listCamera,
  priceFrom: number | null,
  priceTo: number | null
) => {
  let totalProducts: listCamera = [];

  if(priceFrom && !priceTo) {
    totalProducts = products.filter((product) => product.price > priceFrom);
  }
  if(!priceFrom && priceTo) {
    totalProducts = products.filter((product) => product.price < priceTo);
  }
  if(priceFrom && priceTo) {
    totalProducts = products.filter((product) => product.price > priceFrom && product.price < priceTo);
  }

  return totalProducts;
};


export const returnListNumber = (
  value: number
) =>{
  let counter = 0;
  const items = [];

  while(counter <= value) {
    counter += 1;

    items.push(counter);
  }
  return items;
};


export const returnNumber = (number: number) => {
  const valueNumber = (number / 100) * 100;
  let result = 0;

  if(valueNumber > 0 && valueNumber <= 20) {
    result = 1;
  }
  if(valueNumber > 20 && valueNumber <= 40) {
    result = 2;
  }
  if(valueNumber > 40 && valueNumber <= 60) {
    result = 3;
  }
  if(valueNumber > 60 && valueNumber <= 80) {
    result = 4;
  }
  if(valueNumber > 80 && valueNumber > 100) {
    result = 5;
  }

  return result;
};

export const shiftFocus = (evt: KeyboardEvent<HTMLDivElement>, element: MutableRefObject<null | HTMLElement>) => {
  if(evt.key === 'Tab') {
    element.current?.focus();

  }
};
