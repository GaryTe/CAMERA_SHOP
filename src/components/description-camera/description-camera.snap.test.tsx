import {render} from '@testing-library/react';
import DescriptionCamera from './description-camera';
import { product } from '../../util/mocks';

describe('Component: DescriptionCamera', () => {
  it('should render DescriptionCamera', () => {

    const {container} = render(
      <DescriptionCamera
        camera={product}
        onSetMadalWindowAddProduct={jest.fn()}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
