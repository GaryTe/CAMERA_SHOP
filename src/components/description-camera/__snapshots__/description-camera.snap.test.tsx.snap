// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: DescriptionCamera should render DescriptionCamera 1`] = `
<div>
  <div
    class="page-content__section"
  >
    <section
      class="product"
    >
      <div
        class="container"
      >
        <div
          class="product__img"
        >
          <picture>
            <source
              src="img/content/das-auge.webp"
              srcset="img/content/das-auge@2x.webp"
              type="image/webp"
            />
            <img
              alt="Ретрокамера Dus Auge lV"
              height="480"
              src="img/content/das-auge.jpg"
              srcset="img/content/das-auge@2x.jpg"
              width="560"
            />
          </picture>
        </div>
        <div
          class="product__content"
        >
          <h1
            class="title title--h3"
          >
            Ретрокамера Dus Auge lV
          </h1>
          <div
            class="rate product__rate"
          >
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-full-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <svg
              aria-hidden="true"
              height="16"
              width="17"
            >
              <use
                xlink:href="#icon-star"
              />
            </svg>
            <p
              class="visually-hidden"
            >
              Рейтинг: 4
            </p>
            <p
              class="rate__count"
            >
              <span
                class="visually-hidden"
              >
                Всего оценок:
              </span>
              16
            </p>
          </div>
          <p
            class="product__price"
          >
            <span
              class="visually-hidden"
            >
              Цена:
            </span>
            65000
          </p>
          <button
            class="btn btn--purple"
            type="button"
          >
            <svg
              aria-hidden="true"
              height="16"
              width="24"
            >
              <use
                xlink:href="#icon-add-basket"
              />
            </svg>
            Добавить в корзину
          </button>
          <div
            class="tabs product__tabs"
          >
            <div
              class="tabs__controls product__tabs-controls"
            >
              <button
                class="tabs__control is-active"
                type="button"
              >
                Характеристики
              </button>
              <button
                class="tabs__control"
                type="button"
              >
                Описание
              </button>
            </div>
            <div
              class="tabs__content"
            >
              <div
                class="tabs__element is-active"
              >
                <ul
                  class="product__tabs-list"
                >
                  <li
                    class="item-list"
                  >
                    <span
                      class="item-list__title"
                    >
                      Артикул:
                    </span>
                    <p
                      class="item-list__text"
                    >
                      DA4IU67AD5
                    </p>
                  </li>
                  <li
                    class="item-list"
                  >
                    <span
                      class="item-list__title"
                    >
                      Категория:
                    </span>
                    <p
                      class="item-list__text"
                    >
                      Видеокамера
                    </p>
                  </li>
                  <li
                    class="item-list"
                  >
                    <span
                      class="item-list__title"
                    >
                      Тип камеры:
                    </span>
                    <p
                      class="item-list__text"
                    >
                      Коллекционная
                    </p>
                  </li>
                  <li
                    class="item-list"
                  >
                    <span
                      class="item-list__title"
                    >
                      Уровень:
                    </span>
                    <p
                      class="item-list__text"
                    >
                      Нулевой
                    </p>
                  </li>
                </ul>
              </div>
              <div
                class="tabs__element"
              >
                <div
                  class="product__tabs-text"
                >
                  Немецкий концерн BRW разработал видеокамеру Das Auge IV в начале 80-х годов, однако она до сих пор пользуется популярностью среди коллекционеров и яростных почитателей старинной техники.
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
